fsm.go

```diff
@@ -50,12 +50,28 @@ func (r *Raft) runFSM() {
 	var lastIndex, lastTerm uint64
 
 	commit := func(req *commitTuple) {
-		// Apply the log if a command
+		// Apply the log if a command or config change
 		var resp interface{}
-		if req.log.Type == LogCommand {
+		switch req.log.Type {
+		case LogCommand:
 			start := time.Now()
 			resp = r.fsm.Apply(req.log)
 			metrics.MeasureSince([]string{"raft", "fsm", "apply"}, start)
+
+		case LogConfiguration:
+			configStore, ok := r.fsm.(ConfigurationStore)
+			if !ok {
+				// Return early to avoid incrementing the index and term for
+				// an unimplemented operation.
+				if req.future != nil {
+					req.future.respond(nil)
+				}
```

Comment:
Would it be cleaner to set this up in a defer at the start of the closure rather than duplicate it here?